bigtable-hbase-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java

```diff
@@ -63,51 +78,148 @@
   private final BigtableOptions options;
   private final BigtableTableAdminClient bigtableTableAdminClient;
   private BigtableInstanceName bigtableInstanceName;
+  private final TableAdapter2x tableAdapter2x;
 
   public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOException {
     LOG.debug("Creating BigtableAsyncAdmin");
     this.options = asyncConnection.getOptions();
     this.bigtableTableAdminClient = asyncConnection.getSession().getTableAdminClient();
     this.disabledTables = asyncConnection.getDisabledTables();
     this.bigtableInstanceName = options.getInstanceName();
+    this.tableAdapter2x = new TableAdapter2x(options, new ColumnDescriptorAdapter());
   }
 
   @Override
   public CompletableFuture<Void> createTable(TableDescriptor desc, Optional<byte[][]> splitKeys) {
-    throw new UnsupportedOperationException("createTable"); // TODO
+    if (desc.getTableName() == null) {
+      return FutureUtils.failedFuture(new IllegalArgumentException("TableName cannot be null"));
+    }
+
+    return CompletableFuture.supplyAsync(() -> {
```

Comment:
Anything before `FutureUtils.toCompletableFuture(bigtableTableAdminClient.createTableAsync(r.build())))`
should probably be outside of Completable future.  I found the notion of creating the request in a `CompleteableFuture` superfluous.

Reply:
Will remove this pattern. This change can cause exceptions to be thrown out of the CF model which clients will have to account for, but let's see how that will go.

Reply:
Oh... I see.  This is for exception handling.  That makes more sense.  Ok.  You can keep the pattern.  Can you please add a comment somewhere about why this pattern is used?

Reply:
Yes. I just verified not wrapping in CF causes test failures for hbaseLocalMiniClusterTestH2 profile in exception cases. Will leave it, and add some comments.